server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -1789,14 +1805,26 @@ public Node start() throws NodeValidationException {
         );
 
         clusterService.addStateApplier(transportService.getTaskManager());
-        // start after transport service so the local disco is known
-        discovery.start(); // start before cluster service so that it can set initial state on ClusterApplierService
+
+        if (discovery != null) {
+            // start after transport service so the local disco is known
+            discovery.start(); // start before cluster service so that it can set initial state on ClusterApplierService
+        } else {
+            // We're running in clusterless mode, so we manually set the initial state to only include this node.
+            DiscoveryNode localNode = localNodeFactory.getNode();
```

Comment:
wondering if its a bit more maintainable to make a NoOpDiscovery impl / gateway module that manually sets the initial state when started?

Reply:
Lemme give that a try... I might be able to avoid `if`fing out the the various actions.

Reply:
Yeah -- I was able to update that in my latest commit. It is a lot cleaner. Thanks!